### PR TITLE
NEW Created Account::__toString, Account::getFieldsToShow and Account::getAccountNumberOrder to refactor the way account number was shown

### PR DESCRIPTION
**Pull request by marcosgdf**
### Created Account::__toString, Account::getFieldsToShow and Account::getAccountNumberOrder to refactor the way account number was shown.

Because there is much complexity all over Dolibarr to show the bank account number, I considered appropiate to refactor so that the logic is centralized in one function.

To do so I've created the following functions:
- `Account::getFieldsToShow`: to know which fields this bank account needs to display to the user (depending on Dolibarr configuration and on bank account country)
- `Account::getAccountNumberOrder`: returns the fields of the bank account in the order they are going to be displayed (set by the administrator in bank module configuration)
- `Account::__toString`: to show the bank account number just converting the type of the class. I didn't use IBAN because showing IBAN is just showing out the `iban` property of the class. But if you want to, this behaviour can be modified.

`CompanyBankAccount` and `UserBankAccount` classes also benefit from this refactor.

I had to rename `AccountNumber` to `BankAccountNumber` so that the "key" matches the translation key too.

So, to summarize: we now only need to do `(string) $account` to show the bank account number in the appropiate format. If we need to treat data differently, we will use the other functions (like PDF and bank account cards do)

### CODE REVIEW

**Comment by marcosgdf** [comment]
Travis CI build failure has nothing to do with this PR

**Comment by marcosgdf** [comment]
You can merge it now @eldy
